src/app/stores/app_store.ts

```diff
@@ -595,6 +656,21 @@ export class AppStore extends BaseStore {
           }
         }
       }
+      if (this.chart.scaleMappings) {
+        for (const scaleMapping of this.chart.scaleMappings) {
+          if (
+            scaleMapping.expression == expression &&
+            ((scaleMapping.attribute &&
+              scaleMapping.attribute === markAttribute) ||
+              !scaleMapping.attribute)
+          ) {
+            const scaleObject = getById(this.chart.scales, scaleMapping.scale);
+            if (scaleObject && scaleObject.outputType == outputType) {
```

Comment:
Similar to my comment in #203, if charticulator is using a newer version of typescript, you can use optional chaining, this could be `if (scaleObject?.outputType === outputType)`

Reply:
But again, this is no big deal

Reply:
As per comments on #203 charticulator is using an older version of typescript, so optional typing wont work.